src/services/auth.service.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken');
+const CONSTANTS = require('../lib/constants');
+
+const ensureAuthed = (req, res, next) => {
+  // get token from request
+  const token =
+    req.body.token || req.query.token || req.headers['x-access-token'];
+
+  if (token) {
+    jwt.verify(token, CONSTANTS.JWT_ENCRYPTION, (err, decoded) => {
+      if (err) {
+        return res.json({
+          success: false,
+          message: 'Failed to authenticate token.',
+        });
+      }
+      req.decoded = decoded;
+      next();
+      return null;
```

Comment:
The code `return null` won't have any effect since the `next()` will call the next middleware and the the request will finally be sent.